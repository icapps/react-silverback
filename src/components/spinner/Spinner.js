import React from 'react';
import PropTypes from 'prop-types';
import './spinner.scss';

const Spinner = ({ className, spinnerClassName, hasContainer }) => {
  if (hasContainer)
    return (
      <main className={`${className} spinner-container`}>
        <div className={`spinner ${spinnerClassName}`} />
      </main>
    );
  return <div className={`spinner ${spinnerClassName}`} />;
};

Spinner.propTypes = {
  className: PropTypes.string,
  spinnerClassName: PropTypes.string,
  hasContainer: PropTypes.bool,
};

Spinner.defaultProps = {
  className: '',
  spinnerClassName: '',
  hasContainer: true,
};

export default Spinner;
