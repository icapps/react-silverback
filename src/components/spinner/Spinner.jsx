import React from 'react';
import PropTypes from 'prop-types';
import './spinner.css';

const Spinner = props => (
  <main className={`${props.className} spinner-container`}><div className="spinner" /></main>
);

Spinner.propTypes = {
  className: PropTypes.string,
};

Spinner.defaultProps = {
  className: '',
};


export default Spinner;
