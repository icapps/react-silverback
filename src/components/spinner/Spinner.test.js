import React from 'react';
import { shallow, configure } from 'enzyme';
import Spinner from './Spinner';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('Spinner Component', () => {
  it('should render a Spinner component', () => {
    const wrapper = shallow(
      <Spinner
        className="test"
        spinnerClassName="spinner-test"
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Spinner component without a container', () => {
    const wrapper = shallow(
      <Spinner
        hasContainer={false}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
