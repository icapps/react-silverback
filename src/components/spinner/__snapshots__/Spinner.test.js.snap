// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render a Spinner component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    className="test"
    hasContainer={true}
    spinnerClassName="spinner-test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="spinner spinner-test"
/>,
      "className": "test spinner-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "spinner spinner-test",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="spinner spinner-test"
/>,
        "className": "test spinner-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "spinner spinner-test",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Spinner Component should render a Spinner component without a container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    className=""
    hasContainer={false}
    spinnerClassName=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "spinner ",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "spinner ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
